libs/auth/react/src/lib/Authenticator/CheckIdpSession.tsx

```diff
@@ -73,42 +72,55 @@ export const CheckIdpSession = () => {
     />
   )
 
-  const messageHandler = async ({ data }: MessageEvent): Promise<void> => {
-    const sessionInfo = data as SessionInfo
-
-    if (sessionInfo && sessionInfo.message === 'OK') {
-      // SessionInfo was found, check if it is valid or expired
-      if (sessionInfo.isExpired) {
-        return signIn()
-      } else if (sessionInfo.expiresUtc) {
-        // Calculate when the session should expire with padding when we should check again
-        const timeout =
-          new Date(sessionInfo.expiresUtc).getTime() -
-          new Date().getTime() +
-          1000
-
-        if (timeout <= 0) {
-          // The session is expired but for some reason the `isExpired` was not correctly set
-          return signIn()
-        }
+  const messageHandler = useCallback(
+    async ({ data, origin }: MessageEvent): Promise<void> => {
+      if (origin !== authSettings.authority) {
+        return
+      }
```

Comment:
Is there some more checks we can make here to avoid conflicts with other messages?

Reply:
Added type property in the message with specific `SessionInfoResponse` value.